--- Portfile	2018-10-19 21:25:42.624295300 +0200
+++ Portfile-boost	2018-10-20 00:30:41.872155300 +0200
@@ -28,11 +28,7 @@
 checksums       rmd160  efc78b2b1a5d33ed72fcfa6688564c2fcd56772a \
                 sha256  fdfc204fc33ec79c99b9a74944c3e54bd78be4f7f15e260c0e2700a36dc7d3e5
 
-depends_lib     port:zlib \
-                port:expat \
-                port:bzip2 \
-                port:libiconv \
-                port:icu
+depends_lib     port:libiconv
 
 post-extract {
     fs-traverse dir ${workpath} {
@@ -85,14 +81,7 @@
 configure.distcc    no
 
 configure.cmd       ./bootstrap.sh
-configure.args      --without-libraries=python \
-                    --without-libraries=mpi \
-                    --with-icu=${prefix}
-
-if {${os.platform} eq "darwin" && ${os.major} <= 10} {
-    configure.args-append   --without-libraries=context \
-                            --without-libraries=coroutine
-}
+configure.args      --without-icu --with-libraries=chrono,filesystem,program_options,system,thread,test
 
 configure.universal_args
 
@@ -120,18 +109,14 @@
 build.args      -d2 \
                 --layout=tagged \
                 --debug-configuration \
+                --build-type=complete \
                 --user-config=user-config.jam \
-                -sBZIP2_INCLUDE=${prefix}/include \
-                -sBZIP2_LIBPATH=${prefix}/lib \
-                -sEXPAT_INCLUDE=${prefix}/include \
-                -sEXPAT_LIBPATH=${prefix}/lib \
-                -sZLIB_INCLUDE=${prefix}/include \
-                -sZLIB_LIBPATH=${prefix}/lib \
-                -sICU_PATH=${prefix} \
+                -sNO_BZIP2=1 \
+                -sNO_ZLIB=1 \
                 variant=release \
-                threading=single,multi \
-                link=static,shared \
-                -j${build.jobs}
+                threading=multi \
+                link=static \
+                -j2
 
 destroot.cmd            ${worksrcpath}/bjam
 destroot.post_args
@@ -216,23 +201,11 @@
     default_variants +${selected_python}
 }
 
-default_variants +no_single +no_static
-
 variant debug description {Builds debug versions of the libraries as well} {
     build.args-delete   variant=release
     build.args-append   variant=debug,release
 }
 
-variant no_static description {Disable building static libraries} {
-    build.args-delete   link=shared,static
-    build.args-append   link=shared
-}
-
-variant no_single description {Disable building single-threaded libraries} {
-    build.args-delete   threading=single,multi
-    build.args-append   threading=multi
-}
-
 variant regex_match_extra description \
         "Enable access to extended capture information of submatches in Boost.Regex" {
     notes-append "
